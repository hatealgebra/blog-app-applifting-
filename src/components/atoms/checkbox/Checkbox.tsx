import React, { Dispatch } from 'react';
import StyledCheckbox, { StyledCheckboxContainer } from './checkbox.styled';

const Checkbox = ({
  isChecked,
  onChange,
  children,
  isDisabled,
}: CheckboxProps) => {
  return (
    <StyledCheckboxContainer isDisabled={isDisabled}>
      <StyledCheckbox
        disabled={isDisabled}
        onChange={onChange}
        checked={isChecked}
        type="checkbox"
      />
      {children && <span>{children}</span>}
    </StyledCheckboxContainer>
  );
};
export interface CheckboxProps {
  isChecked: boolean;
  onChange: any;
  children?: string;
  isDisabled?: boolean;
}

export default Checkbox;
