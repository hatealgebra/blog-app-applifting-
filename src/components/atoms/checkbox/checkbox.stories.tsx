import { action } from '@storybook/addon-actions';
import React from 'react';
import Checkbox from './Checkbox';

const setCheck = () => action('Clicked!');

export const CheckboxExample = () => {
  return (
    <>
      <Checkbox isChecked={false} onChange={setCheck} />
      <Checkbox isChecked={true} onChange={setCheck} />
      <Checkbox isDisabled isChecked={false} onChange={setCheck} />
      <Checkbox isDisabled isChecked={true} onChange={setCheck} />
    </>
  );
};
export const CheckboxWithLabel = () => {
  return (
    <>
      <Checkbox isChecked={false} onChange={setCheck}>
        Checkbox with label
      </Checkbox>
      <Checkbox isDisabled isChecked={false} onChange={setCheck}>
        Checkbox with label
      </Checkbox>
    </>
  );
};

export default {
  component: Checkbox,
  title: 'Atoms/Checkbox',
};
