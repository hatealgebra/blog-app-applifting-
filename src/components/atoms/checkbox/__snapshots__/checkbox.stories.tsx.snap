// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Checkbox CheckboxExample smoke-test 1`] = `
<div class="sc-ftLKQv iibJaj">
  <input type="checkbox"
         class="sc-iBAaJG ciwLrW"
  >
</div>
<div class="sc-ftLKQv iibJaj">
  <input type="checkbox"
         class="sc-iBAaJG ciwLrW"
         checked
  >
</div>
<div class="sc-ftLKQv jAoYjG">
  <input disabled
         type="checkbox"
         class="sc-iBAaJG ciwLrW"
  >
</div>
<div class="sc-ftLKQv jAoYjG">
  <input disabled
         type="checkbox"
         class="sc-iBAaJG ciwLrW"
         checked
  >
</div>
`;

exports[`Atoms/Checkbox CheckboxWithLabel smoke-test 1`] = `
<div class="sc-ftLKQv iibJaj">
  <input type="checkbox"
         class="sc-iBAaJG ciwLrW"
  >
  <span>
    Checkbox with label
  </span>
</div>
<div class="sc-ftLKQv jAoYjG">
  <input disabled
         type="checkbox"
         class="sc-iBAaJG ciwLrW"
  >
  <span>
    Checkbox with label
  </span>
</div>
`;
